### ISSUE TITLE
Incompatible version of @date-io/core in dependencies & peerDependencies #47

### ISSUE THREAD

**Issue by FedeBev** [material-table-core/core#169]
**Describe the bug**
`@material-table/core` requires `@date-io/core@^2.10.6`, however `@material-table/core` also has `@material-ui/pickers@3.3.10` dependency with has `@date-io/core@^1.3.6` as peer.

**To Reproduce**
If you install `@material-table/core@2.3.32` and `@date-io/core@^2.10.6` you'll see the following warning
```text
warning "@material-table/core > @material-ui/pickers@3.3.10" has incorrect peer dependency "@date-io/core@^1.3.6".
```

If you install `@material-table/core@2.3.32` and `@date-io/core@^1.3.6` you'll see the following warning
```text
warning " > @material-table/core@2.3.32" has incorrect peer dependency "@date-io/core@^2.10.6"
```

**Expected behavior**
No warnings

**Possible solutions**
Since `@material-ui/pickers`[ is no longer supported](https://github.com/mui-org/material-ui-pickers/issues/2157), I guess we can:
- Downgrade the version of `@date-io/core`
- [MUI v5 should get a better support to date-time pickers](https://next.material-ui.com/components/pickers/ ) (right now is a bit sour), but this issue can be addressed while workign on MUI 5 compatibility (#55)

**Comment by matthewoestreich** [comment]
Thank you. This is especially odd because there was another issue for this which I resolved last night. It would seem v2.x has made it back. I will take a look here in a min (if you want to shoot over a PR real quick I can merge it as well)

Thank you!

**Comment by matthewoestreich** [comment]
Shouldn't downgrading the pickers library to a version that uses a date-io/core version that is compatible? Thoughts?

**Comment by FedeBev** [comment]
> Shouldn't downgrading the pickers library to a version that uses a date-io/core version that is compatible? Thoughts?

Do you mean we shouldn't downgrade date-io/core? Otherwise I think I don't get it.

**Comment by matthewoestreich** [comment]
Sorry that's my fault. We just need to downgrade date-io/core, right? 

I recently downgraded date-io/date-fns and I confused the two.

**Comment by FedeBev** [comment]
> We just need to downgrade date-io/core, right?

I guess that's the easiest solution right now, just keep in mind that this issue is going to come up again with MUI 5

I can deal with this within a few hours and crate a PR if you want

**Comment by matthewoestreich** [comment]
No doubt. That should be in the next major version (worrying about MUI5).

v2.3.33 has been published with @date-io/core at v1.3.13. Please let me know if this is still an issue for you.
